Libraries/pbxbuild/Sources/Phase/ProductTypeResolver.cpp

```diff
@@ -268,6 +269,135 @@ ResolveFrameworkStructure(Phase::Environment const &phaseEnvironment, Phase::Con
     return true;
 }
 
+static ext::optional<std::string>
+FindUmbrellaHeaderName(pbxsetting::Environment const &environment, std::string const &moduleName, pbxproj::PBX::Target::shared_ptr const &target)
+{
+    std::string lowerModuleName;
+    std::transform(moduleName.begin(), moduleName.end(), std::back_inserter(lowerModuleName), ::tolower);
+
+    ext::optional<std::string> umbrellaHeader;
+    for (pbxproj::PBX::BuildPhase::shared_ptr const &buildPhase : target->buildPhases()) {
+        if (buildPhase->type() != pbxproj::PBX::BuildPhase::Type::Headers) {
+            continue;
+        }
+
+        for (pbxproj::PBX::BuildFile::shared_ptr const &buildFile : buildPhase->files()) {
+            if (buildFile->fileRef() == nullptr) {
+                continue;
+            }
+
+            std::string filePath = environment.expand(buildFile->fileRef()->resolve());
+            if (FSUtil::GetFileExtension(filePath) == "h") {
+                std::string baseName = FSUtil::GetBaseNameWithoutExtension(filePath);
+
+                std::string lowerBaseName;
+                std::transform(baseName.begin(), baseName.end(), std::back_inserter(lowerBaseName), ::tolower);
+
+                if (lowerBaseName == lowerModuleName) {
+                    return baseName;
+                }
+            }
+        }
+    }
+
+    return ext::nullopt;
+}
+
+static bool
+ResolveModuleMap(Phase::Environment const &phaseEnvironment, Phase::Context *phaseContext, pbxsetting::Environment const &environment)
```

Comment:
Pointers are for anything mutable, so at the call site you can see it can be mutated (i.e. the `&` for the pointer) rather than being identical to passing by value. In this case `Phase::Context` holds a `Tool::Context` which has a mutable `invocations` vector that this adds to, so it's a pointer to note that it's being mutated.